### ISSUE TITLE
[Sluggable] Slug is not unique on creation

### ISSUE THREAD

**Issue by hermann8u** [KnpLabs/DoctrineBehaviors#482]
Hello,

I spot a bug on unique slugs generation when the "getRegenerateSlugOnUpdate" return false.

If $sluggable->getId() return null (most of the time on creation), the result of the query will be 0.

https://github.com/KnpLabs/DoctrineBehaviors/blob/1bcce5a9025eca9fab6e4783b544a4dfcf894b9e/src/Repository/DefaultSluggableRepository.php#L26-L33

Any idea how to fix this?

**Comment by TomasVotruba** [comment]
Hi,

could you provide minimal failing test case?

**Comment by hermann8u** [comment]
---- TEST CASE ----
The following entity will have the same slug if you create two of them with the same title:

``` php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Knp\DoctrineBehaviors\Contract\Entity\SluggableInterface;

/**
 * @ORM\Entity()
 * @ORM\Table()
 */
class NewsSlug implements SluggableInterface
{
    use SluggableTrait;

    /**
     * @ORM\Id
     * @ORM\Column(type="integer", unique=true)
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @ORM\Column(type="string")
     */
    private $title;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): ?string
    {
        return $this->title;
    }

    public function setTitle($title): self
    {
        $this->title = $title;

        return $this;
    }

    /**
     * @inheritDoc
     */
    public function getSluggableFields(): array
    {
        return ['title'];
    }

    public function shouldGenerateUniqueSlugs(): bool
    {
        return true;
    }
}
```
---- END TEST CASE ----

I was wrong when I said that this occurs when the "getRegenerateSlugOnUpdate" return false.

Concerning the DoctrineBehaviors/src/Repository/DefaultSluggableRepository.php :

The condition on the id at line 30 is here to handle update but should not be apply on creation.
The easiest way to fix it is to add dynamically the line 30 and 32 when the result of $sluggable->getId() is not null.

I'll try to make the PR today if I find the time.

**Comment by TomasVotruba** [comment]
> I'll try to make the PR today

Awesome. I'll review ASAP

**Comment by hermann8u** [comment]
This issue has been fixed in PR #487

**Comment by TomasVotruba** [comment]
Tip for automated closing on PR merge: https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords


tl;dr; Add this to PR body description

```
Closes #1 
```
